openmc/data/neutron.py

```diff
@@ -0,0 +1,926 @@
+
+        # Add normal and redundant reactions
+        for mt in data.reactions:
+            if mt in self:
+                self[mt].xs[strT] = data[mt].xs[strT]
+            else:
+                warn("Tried to add cross sections for MT={} at T={} but this "
+                     "reaction doesn't exist.".format(mt, strT))
+
+        # Add probability tables
+        if strT in data.urr:
+            self.urr[strT] = data.urr[strT]
+
+    def add_elastic_0K_from_endf(self, filename, overwrite=False):
+        """Append 0K elastic scattering cross section from an ENDF file.
+
+        Parameters
+        ----------
+        filename : str
+            Path to ENDF file
+        overwrite : bool
+            If existing 0 K data is present, this flag can be used to indicate
+            that it should be overwritten. Otherwise, an exception will be
+            thrown.
+
+        Raises
+        ------
+        ValueError
+            If 0 K data is already present and the `overwrite` parameter is
+            False.
+
+        """
+        # Check for existing data
+        if '0K' in self.energy and not overwrite:
+            raise ValueError('0 K data already exists for this nuclide.')
+
+        data = type(self).from_endf(filename)
+        if data.resonances is not None:
+            x = []
+            y = []
+            for rr in data.resonances:
+                if isinstance(rr, res.RMatrixLimited):
+                    raise TypeError('R-Matrix Limited not supported.')
+                elif isinstance(rr, res.Unresolved):
+                    continue
+
+                # Get energies/widths for resonances
+                e_peak = rr.parameters['energy'].values
+                if isinstance(rr, res.MultiLevelBreitWigner):
+                    gamma = rr.parameters['totalWidth'].values
+                elif isinstance(rr, res.ReichMoore):
+                    df = rr.parameters
+                    gamma = (df['neutronWidth'] +
+                             df['captureWidth'] +
+                             abs(df['fissionWidthA']) +
+                             abs(df['fissionWidthB'])).values
+
+                # Determine peak energies and widths
+                e_min, e_max = rr.energy_min, rr.energy_max
+                in_range = (e_peak > e_min) & (e_peak < e_max)
+                e_peak = e_peak[in_range]
+                gamma = gamma[in_range]
+
+                # Get midpoints between resonances (use min/max energy of
+                # resolved region as absolute lower/upper bound)
+                e_mid = np.concatenate(
+                    ([e_min], (e_peak[1:] + e_peak[:-1])/2, [e_max]))
+
+                # Add grid around each resonance that includes the peak +/- the
+                # width times each value in _RESONANCE_ENERGY_GRID. Values are
+                # constrained so that points around one resonance don't overlap
+                # with points around another. This algorithm is from Fudge.
```

Comment:
Optional: I would clarify the statement `This algorithm is from Fudge`.  My guess is Fudge is either a paper or a textbook, but it is not entirely clear from context.